src/bytecode/astgen.rs

```diff
@@ -284,9 +289,57 @@ impl<'a, 'ast> AstBytecodeGen<'a, 'ast> {
         dest
     }
 
+    fn visit_expr_assert(&mut self, expr: &ExprCallType, _dest: DataDest) {
+        let lbl_assert = self.gen.create_label();
+
+        let assert_reg = self.visit_expr(&*expr.args[0], DataDest::Alloc);
+
+        self.gen.emit_jump_if_true(assert_reg, lbl_assert);
+
+        let cls_id = self.vm.vips.error_class;
+        let cls = self.vm.classes.idx(cls_id);
+        let cls = cls.read();
+
+        let fct_id = cls.constructor.expect("constructor is missing");
+        let fct = self.vm.fcts.idx(fct_id);
+        let fct = fct.read();
+
+        let arg_types = fct
+            .params_with_self()
+            .iter()
+            .map(|&arg| {
+                self.specialize_type_for_call(
+                    &CallType::CtorNew(cls_id, fct_id, TypeParams::Empty),
```

Comment:
I don't know, if it is correct to pass `TypeParams::Empty` to the function. What do you think?

Reply:
That should be fine, the error should not be generic.